feat: add filtered domains in filter in tools dashboard

janvi01 opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow our contribution guidelines
2. Test your changes and attach their results to the pull request.
3. Update the relevant documentation.
-->

**Description**

The PR adds the filtered domain below the dropdown in the tool's dashboard. Below is the screenshot that depicts the implemenatation.

Screenshot -

<img width="1791" alt="Screenshot 2023-01-28 at 11 48 38 AM" src="https://user-images.githubusercontent.com/34344234/215250976-c55e4d7f-0c81-4486-af19-ea4b0bf3f822.png">


**Related issue(s)**

Fixes #1270 
<!-- If you refer to a particular issue, provide its number, otherwise, remove this section.
For example, `Resolves #123`, `Fixes #43`, or `See also #33`. The `See also #33` option will not automatically close the issue after the PR merge. -->

janvi01 committed:
added filtered domains in filter in tools dashboard

akshatnema reviewed (changes requested):
![image](https://user-images.githubusercontent.com/76521428/215323132-04ebd85d-7385-41fb-8af6-ad0828ce2814.png)

Hey @janvi01, thanks for contributing to the issue. My suggestion is that if we are mentioning the tags selected in the filter, let's not just make it only UI. We should make it some way useful. Add a cross (in icon) button beside each tag so as to remove the filter without opening the dropdown for the field. Also, I find that these values look slightly similar to the fields given inside the Filter Dropdown. So, Can you please make it like having light gray border and on hover, slightly dark to make them different from others.

![image](https://user-images.githubusercontent.com/76521428/215323798-08c2c2ca-8279-45f8-b669-7d233d4acbc4.png)

It will be nice if you can add a feature that when some tags are selected in the filter, this filter header shows the number of tags selected like "3 options selected", and none of them are selected, "Select Languages..." are there. I hope you will understand. If not, you can ask me freely.

janvi01 commented:
Hey, @akshatnema Thanks for the amazing improvement suggestions. I will update the PR soon.

janvi01 committed:
add remove filter tags functionality, enhance tags ui

janvi01 commented:
I have changed the tags ui as per the suggestion, added the functionality to remove the filter and labels in the dropdown when a certain number of tags are selected.

Also when 1 option is selected, it says 'option' else 'options'.

akshatnema commented on components/tools/Filters.js:
```diff
@@ -47,6 +47,38 @@ export default function Filters({setOpenFilter}) {
+        })}
+      </div>
+    );
+  };
```
Kindly make a separate file for this component. You should dispatch the state change functions if required in order to get values changed using the close button. There shouldn't be any component inside a component.

akshatnema commented on components/tools/Filters.js:
```diff
@@ -47,6 +47,38 @@ export default function Filters({setOpenFilter}) {
+  }
+
+  // function to display filtered domain
+  const FilterDisplay = (props) => {
```
```suggestion
  const FilterDisplay = ({ checkedDomain = [], setDomain }) => {
```

akshatnema commented on components/tools/Filters.js:
```diff
@@ -47,6 +47,38 @@ export default function Filters({setOpenFilter}) {
+            <div
+              key={index}
+              className={twMerge(
+                `hover:bg-neutral-600 border border-gray-600 text-gray-600 p-1 pb-0 rounded-2xl flex gap-1 cursor-pointer items-start `
```
```suggestion
                `hover:border-black border border-gray-600 text-gray-600 hover:text-black p-1 pb-0 rounded-2xl flex gap-1 items-start`
```

akshatnema commented on components/tools/Filters.js:
```diff
@@ -47,6 +47,38 @@ export default function Filters({setOpenFilter}) {
+              )}
+            >
+              <div className="text-xs -mt-[1px] mb-[1px]">{items}</div>
+              <button className="hover:bg-gray-100 p-0.5 rounded-full" onClick={(e) => handleClickOption(e, items, props.checkedDomain, props.setDomain)}>
```
```suggestion
              <button className="hover:bg-gray-100 p-1 rounded-full -mt-[2px]" onClick={(e) => handleClickOption(e, items, props.checkedDomain, props.setDomain)}>
```

akshatnema commented on components/tools/Filters.js:
```diff
@@ -47,6 +47,38 @@ export default function Filters({setOpenFilter}) {
+                `hover:bg-neutral-600 border border-gray-600 text-gray-600 p-1 pb-0 rounded-2xl flex gap-1 cursor-pointer items-start `
+              )}
+            >
+              <div className="text-xs -mt-[1px] mb-[1px]">{items}</div>
```
```suggestion
              <div className="text-xs m-auto h-fit">{items}</div>
```

akshatnema reviewed (changes requested):
Certain changes are needed in the PR.

janvi01 committed:
add: separate new file for filter tags display

akshatnema commented on components/tools/FiltersDisplay.js:
```diff
@@ -0,0 +1,32 @@
+import { twMerge } from "tailwind-merge";
+
+export default function FiltersDisplay({ checkedDomain = [], setDomain }) {
```
```suggestion
export default function FiltersDisplay({ checkedValues = [], setValues }) {
```

Kindly make the name of the variables according to the use/functionality of them in the component.

janvi01 committed:
fix: variable names

akshatnema committed:
Merge branch 'master' into toolsFilter

akshatnema commented:
/rtm
